core/lib/spree/core/product_duplicator.rb

```diff
@@ -72,7 +72,7 @@ def reset_properties
     end
 
     def sku_generator(sku)
-      "COPY OF #{Variant.where('sku like ?', "%#{sku}").order(:created_at).last.sku}"
+      "COPY OF #{Variant.with_deleted.where('sku like ?', "%#{sku}").order(:created_at).last.sku}"
```

Comment:
@manishrawat0212  This would raise an exception in case there's no variant from the last method call